src/model.jl

```diff
@@ -116,43 +119,112 @@ function create_model(
+    flow_sum_indexes_outgoing_lowest =
+        Dict{TupleAssetRPTimeBlock,Vector{TupleDurationFlowTimeBlock}}()
+    flow_sum_indexes_incoming_highest =
+        Dict{TupleAssetRPTimeBlock,Vector{TupleDurationFlowTimeBlock}}()
+    flow_sum_indexes_outgoing_highest =
+        Dict{TupleAssetRPTimeBlock,Vector{TupleDurationFlowTimeBlock}}()
+    for a ∈ A, rp ∈ RP
+        for B ∈ Pl[(a, rp)]
+            flow_sum_indexes_incoming_lowest[a, rp, B] = TupleDurationFlowTimeBlock[]
+            flow_sum_indexes_outgoing_lowest[a, rp, B] = TupleDurationFlowTimeBlock[]
+            for u ∈ inneighbor_labels(graph, a)
+                add_to_flow_sum_indexes!(
+                    flow_sum_indexes_incoming_lowest[a, rp, B],
+                    a,
+                    rp,
+                    B,
+                    (u, a),
+                )
+            end
+
+            for v ∈ outneighbor_labels(graph, a)
+                add_to_flow_sum_indexes!(
+                    flow_sum_indexes_outgoing_lowest[a, rp, B],
+                    a,
+                    rp,
+                    B,
+                    (a, v),
+                )
+            end
+        end
+        for B ∈ Ph[(a, rp)]
+            flow_sum_indexes_incoming_highest[a, rp, B] = TupleDurationFlowTimeBlock[]
+            flow_sum_indexes_outgoing_highest[a, rp, B] = TupleDurationFlowTimeBlock[]
+            for u ∈ inneighbor_labels(graph, a)
+                add_to_flow_sum_indexes!(
+                    flow_sum_indexes_incoming_highest[a, rp, B],
+                    a,
+                    rp,
+                    B,
+                    (u, a),
+                )
+            end
+
+            for v ∈ outneighbor_labels(graph, a)
+                add_to_flow_sum_indexes!(
+                    flow_sum_indexes_outgoing_highest[a, rp, B],
+                    a,
+                    rp,
+                    B,
+                    (a, v),
+                )
+            end
+        end
+    end
+
     ## Expressions for balance constraints
     @expression(
         model,
         incoming_flow_lowest_resolution[a ∈ A, rp ∈ RP, B ∈ Pl[(a, rp)]],
         sum(
-            duration(B, B_flow, rp) * flow[(u, a), rp, B_flow] for
-            u in inneighbor_labels(graph, a), B_flow ∈ graph[u, a].partitions[rp] if
-            B_flow[end] ≥ B[1] && B[end] ≥ B_flow[1]
+            d * flow[f, rp, B_flow] for
+            (d, f, B_flow) ∈ flow_sum_indexes_incoming_lowest[(a, rp, B)]
         )
     )
 
     @expression(
         model,
         outgoing_flow_lowest_resolution[a ∈ A, rp ∈ RP, B ∈ Pl[(a, rp)]],
         sum(
-            duration(B, B_flow, rp) * flow[(a, v), rp, B_flow] for
-            v in outneighbor_labels(graph, a), B_flow ∈ graph[a, v].partitions[rp] if
-            B_flow[end] ≥ B[1] && B[end] ≥ B_flow[1]
+            d * flow[f, rp, B_flow] for
+            (d, f, B_flow) ∈ flow_sum_indexes_outgoing_lowest[(a, rp, B)]
         )
     )
```

Comment:
If this is so similar for incoming and outgoing, couldn't we generalize?

Reply:
There is a way, but I would need to deconstruct the `@expression` into something else much less readable.